scripts/common.make

```diff
@@ -43,6 +64,12 @@ else # Not building for Windows
   DLEXT ?= so
   EXEXT=
   SHARED_SUFFIX=
+  ifndef THREADING
+    # Auto-detect configurations with pthread.
+    ifeq (control,$(call remove_unset_options,control MBEDTLS_THREADING_C MBEDTLS_THREADING_PTHREAD))
+      THREADING := pthread
```

Comment:
Why are we setting this when we have neither MBEDTLS_THREADING_C nor MBEDTLS_THREADING_PTHREAD in the configuration?

(The CI is happily passing, so this is surely correct. I just would like to know what I am missing here.)

Reply:
Yep, I misunderstood what the remove_unset_options function does.

Reply:
Yes, I don't think this is obvious at all